File: go/create_add_on_request.go
// this file is @generated
package meteroid

type CreateAddOnRequest struct {
	Description *string `json:"description,omitempty"`

	MaxInstancesPerSubscription *int32 `json:"max_instances_per_subscription,omitempty"`

	Name string `json:"name"`

	PriceId PriceId `json:"price_id"`

	ProductId ProductId `json:"product_id"`

	SelfServiceable *bool `json:"self_serviceable,omitempty"`
}

File: python/meteroid/models/create_add_on_request.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel
from .price_id import PriceId
from .product_id import ProductId


@dataclasses.dataclass
class CreateAddOnRequest(BaseModel):
    name: str

    price_id: PriceId

    product_id: ProductId

    description: t.Optional[str] = None

    max_instances_per_subscription: t.Optional[int] = None

    self_serviceable: t.Optional[bool] = None
